sdk/tables/Azure.Data.Tables/src/Extensions/DictionaryTableExtensions.cs

```diff
@@ -159,8 +159,8 @@ internal static T ToTableEntity<T>(this IDictionary<string, object> entity, Prop
             {typeof(byte[]), (property, propertyValue, result) =>  property.SetValue(result, Convert.FromBase64String(propertyValue as string))},
             {typeof(long), (property, propertyValue, result) =>  property.SetValue(result, long.Parse(propertyValue as string, CultureInfo.InvariantCulture))},
             {typeof(long?), (property, propertyValue, result) =>  property.SetValue(result, long.Parse(propertyValue as string, CultureInfo.InvariantCulture))},
-            {typeof(double), (property, propertyValue, result) =>  property.SetValue(result, (double)propertyValue)},
-            {typeof(double?), (property, propertyValue, result) =>  property.SetValue(result, (double?)propertyValue)},
+            {typeof(double), (property, propertyValue, result) =>  property.SetValue(result, propertyValue)},
+            {typeof(double?), (property, propertyValue, result) =>  property.SetValue(result, propertyValue)},
```

Comment:
This change is unrelated to the unit tests, but is a random fix discovered while experimenting with running tests against cosmos endpoints.

Given a double value with no decimal, such as 1, storage sends the value as "1.0" whereas Cosmos sends it as "1". The storage response format ensures that the type can always be implicitly determined but the Cosmos format does not. This caused a boxed casting exception here - the implicit cast resolves the problem.

Note this can still cause some type ambiguity when dealing with dictionary<string, object> entities.

Reply:
> Given a double value with no decimal, such as 1, storage sends the value as "1.0" whereas Cosmos sends it as "1". The storage response format ensures that the type can always be implicitly determined but the Cosmos format does not. This caused a boxed casting exception here - the implicit cast resolves the problem.

We should have an issue with a set of protocol differences you've found.

Reply:
tracked in #12426